### PR TITLE
Vdirsyncer 0.11.0

### PR DESCRIPTION
**Pull request by benjaminfrank**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

Includes upstream launchd .plist

### CODE REVIEW

File: Formula/vdirsyncer.rb [hunk 1]
```diff
@@ -67,6 +67,27 @@ def install
 
     bin.install Dir["#{libexec}/bin/*"]
     bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+
+    prefix.install "contrib"
```
Comment (benjaminfrank) [inline Formula/vdirsyncer.rb]: This is my kind of workaround so I can read the upstream .plist later in ```def plist```. I tried using instance variables but that did not work. Class variables produced an empty .plist file.
Any suggestions on how this part can be improved?

**Comment by DomT4** [comment]
@BrewTestBot test this please

File: Formula/vdirsyncer.rb [hunk 2]
```diff
@@ -67,6 +67,27 @@ def install
...
+  def caveats; <<-EOS.undent
+      The sync intervall in the launchd .plist file is set to #{SYNC_INTERVAL}s
+      and should be adjusted to your when enabled.
+    EOS
+  end
+
+  plist_options :manual => "vdirsyncer"
+
+  def plist
+    p = File.read(prefix/"contrib/vdirsyncer.plist")
```
Comment (MikeMcQuaid) [inline Formula/vdirsyncer.rb]: Instead use `prefix.install "contrib/vdirsyncer.plist"` above and then use `inreplace` on that file.

File: Formula/vdirsyncer.rb [hunk 3]
```diff
@@ -67,6 +67,27 @@ def install
...
+    prefix.install "contrib"
+  end
+
+  SYNC_INTERVAL = 60
+
+  def caveats; <<-EOS.undent
+      The sync intervall in the launchd .plist file is set to #{SYNC_INTERVAL}s
+      and should be adjusted to your when enabled.
+    EOS
+  end
```
Comment (MikeMcQuaid) [inline Formula/vdirsyncer.rb]: Don't need these caveats.

File: Formula/vdirsyncer.rb [hunk 4]
```diff
@@ -67,6 +67,27 @@ def install
 
     bin.install Dir["#{libexec}/bin/*"]
     bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+
+    prefix.install "contrib"
+  end
+
+  SYNC_INTERVAL = 60
```
Comment (MikeMcQuaid) [inline Formula/vdirsyncer.rb]: Just embed this in the `inreplace`.

File: Formula/vdirsyncer.rb (hunk 2 above)
Comment (MikeMcQuaid) [inline Formula/vdirsyncer.rb]: The `inreplace` for `LOCALE` should be in `post_install`.

**Comment by MikeMcQuaid** [comment]
Installation is failing: http://bot.brew.sh/job/Homebrew%20Core%20Pull%20Requests/2382/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/install_vdirsyncer/

**Comment by benjaminfrank** [comment]
Not sure why the installation fails, maybe related to the plist file?

**Comment by benjaminfrank** [comment]
```
Warning: The post-install step did not complete successfully
You can try again using `brew postinstall vdirsyncer`
```
Any additional info I could get on what the issue is? The only thing that comes to my mind is that `ENV['LANG']` is not defined somehow

**Comment by DomT4** [comment]
I believe `LANG` is unset by default on OS X?

**Comment by benjaminfrank** [comment]
Hm, any hint how to get the locale then? Otherwise I could set the locale via env params in the .plist itself and then tell vdirsyncer about it...but that sounds somewhat strange. I'd rather get whatever locale is default on osx.

**Comment by UniqMartin** [comment]
What is the purpose of setting the locale in the `.plist` file? It feels a bit arbitrary to hard-code this value to whatever the user happened to have configured during installation. I would tend to use a fixed, but safe default instead (like `en_US.UTF-8`).

On a side note, the `.plist` file sets both `LANG` and `LC_ALL` and this is pretty pointless as they are both essentially the same setting, just with different priorities. `LANG` is used for all `LC_*` values that haven't been explicitly set and `LC_ALL` overrides all `LC_*` values, even if `LANG` or individual `LC_*` variables have been set to something else. (At least that's my understanding of the locale system. You can try setting and/or unsetting various combinations of these and then run `locale` to see the effect.)

**Comment by benjaminfrank** [comment]
Vdirsyncer requires a proper locale set, that is locale != C. Hence I need to set the locale to whatever is the default of the system. it seems getting `ENV["LANG"]` is not how to do that. I'll try to find how to do this. Any hint is very welcome :)

about LC_ALL / LANG, you'r right, it should suffice to set one of them.

**Comment by benjaminfrank** [comment]
Sadly, all of the thing I found to get the  locale return empty when brew install runs:
- osascript -e user locale of (get system info)
- defaults read .GlobalPreferences AppleLocale
However, I get the feeling that osx simply supports many locales anyway, so using something like `en_US.UTF-8`should just work. not sure what this would mean for foreign characters in the application files though

**Comment by UniqMartin** [comment]
I'm not sure if I can provide a good suggestion, but both the OS X Terminal and the somewhat popular iTerm2 have an option named “Set locale environment variables on startup”. With this unchecked, none of the `LANG` and `LC_*` are set and a plain `locale` is reporting the C locale for all categories. With it checked, both set `LANG` to whatever best matches the user's configured locale on OS X, which happens to be `en_US.UTF-8` on my system. I think your best bet is to look at the output of `locale` (for the category that best matches the use case) and then take that or some fallback if this happens to be the C locale. E.g. here is my output of `locale`:

```
$ locale
LANG="en_US.UTF-8"
LC_COLLATE="C"
LC_CTYPE="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_ALL=
```

Which of those lines would you tend to use? (Notice that I don't use `en_US.UTF-8` for everything.) The question you'd need know how to answer is what this locale information is used for. (I'm not familiar with it, so I cannot answer this myself.)

**Comment by MikeMcQuaid** [comment]
I think relying on the system variable or falling back to `en_US.UTF-8` is probably a reasonable default 👍

**Comment by UniqMartin** [comment]
I agree. That's probably the best solution without over-complicating things. In Ruby code this would be something like `ENV["LC_ALL"] || ENV["LANG"] || "en_US.UTF-8"` if you wanted to still respect that `LC_ALL` takes precedence over `LANG`.

**Comment by benjaminfrank** [comment]
Ok, I incorporated your suggestions, thanks for the help! I also added some non-western chars and test for them, just to make sure the locale is a sensible choice. locally it works as it should now: pick the locale the user has set and fall back to en_US.UTF-8 (which still supports chinese/arabic/etc chars as far as I understand)

**Comment by UniqMartin** [comment]
> (which still supports chinese/arabic/etc chars as far as I understand)

Yes, every `.UTF-8` locale supports this (and there are hardly any non-UTF-8 locales in use nowadays), as the displayable characters really only depend on the character encoding used, not on the selected language. The language itself then affects sorting order, date/time formatting, and a few other things.

**Comment by MikeMcQuaid** [comment]
And if you haven't set a locale I think it's totally reasonable for us to default to English (like most other software).

**Comment by MikeMcQuaid** [comment]
Thanks again @benjaminfrank for this PR, your previous contributions and your patience and hard work getting this merged 👏

File: Formula/vdirsyncer.rb [hunk 5]
```diff
@@ -1,8 +1,9 @@
+# coding: utf-8
```
Comment (DomT4) [inline Formula/vdirsyncer.rb]: Minor belated notation, this shouldn't be necessary any more. We've logic rammed into the core to treat formulae files as UTF-8.

Comment (benjaminfrank) [inline Formula/vdirsyncer.rb]: That is most likely a leftover from me simply copying some other Formula as basis for this one. I'll try to remember to remove it in the next version bump

Comment (DomT4) [inline Formula/vdirsyncer.rb]: I'll squash it out of existence in a minute. Was just an FYI really 😺

Comment (DomT4) [inline Formula/vdirsyncer.rb]: https://github.com/Homebrew/homebrew-core/commit/e63d284c793debd8b5011973caf7060a635c0c5b

**Pull request closed without merging**